### PR TITLE
reports & rerun require TestRunFinishedEvent to be posted

### PR DESCRIPTION
**Pull request by kaskasi**
TL;DR - In 4.2.4 TestRunFinished Event is not posted on the EventBus which breaks reporting and rerun features.

Longer Explanation: Previous to 4.2.2 `CucumberExecutor` posted `TestRunFinished` during the `execute` method. In 4.2.4  `CucumberJunitRunner` is not posting that event on the EventBus anymore. All Formatters like `RerunFormatter` or `HTMLFormatter` only write the test results to disk once they receive `TestRunFinished`. This leads to rerun file or reports always being empty after test run.

I propose the following change to fix this Bug. In `CucumberJunitRunner` similar to `Cucumber` class from core I would override `childrenInvoker` like this

```   
   @Override
    protected Statement childrenInvoker(RunNotifier notifier) {
        final Statement features = super.childrenInvoker(notifier);
        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                features.evaluate();
                bus.send(new TestRunFinished(bus.getTime()));
            }
        };
    }
```
This change also requires to make `EventBus bus` a class wide variable, that can as previously be initialized during constructor.

Please let me know what you think about my suggestion.

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
Hi @kaskasi,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber
